zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/record/value/JobListenerEventType.java

```diff
@@ -33,5 +33,39 @@ public enum JobListenerEventType {
    * the listener should be triggered at the end of an execution, such as the completion of a
    * process instance, sub-process or element (task, event, gateway).
    */
-  END
+  END,
+
+  /**
+   * Represents the `create` event for a task listener. This event type is used to indicate that the
+   * listener should be triggered when a user task is created. This occurs when the task becomes
+   * available in the workflow for the first time.
+   */
+  CREATE,
+
+  /**
+   * Represents the `ASSIGNMENT` event for a task listener. This event type is used to indicate that
+   * the listener should be triggered when a user task is assigned, claimed (assigned to the current
+   * user), or unassigned.
+   */
+  ASSIGNMENT,
+
+  /**
+   * Represents the `update` event for a task listener. This event type is used to indicate that the
+   * listener should be triggered when a user task's properties are updated. This includes changes
+   * to attributes like `priority`, `due date` or other task metadata.
+   */
+  UPDATE,
+
+  /**
+   * Represents the `complete` event for a task listener. This event type is used to indicate that
+   * the listener should be triggered when a user task is completed.
+   */
+  COMPLETE,
+
+  /**
+   * Represents the `cancel` event for a task listener. This event type is used to indicate that the
+   * listener should be triggered when a user task is canceled or terminated before completion. This
+   * can occur due to workflow interruptions or exceptions.
+   */
+  CANCEL
```

Comment:
🔧 If you want to reduce the size of the PR, you could extract these and add them later again

Reply:
Thanks! I've removed the other job listener event types except for `COMPLETE`, as they aren't required at this stage. We'll reintroduce them in dedicated PRs when we'll work on support for `update/assign/etc.` task listeners.